src/sentry/incidents/logic.py

```diff
@@ -1286,18 +1292,19 @@ def get_alert_rule_trigger_action_pagerduty_service(
 
 
 def get_alert_rule_trigger_action_sentry_app(organization, sentry_app_id):
-    try:
-        # query for the sentry app but make sure it's installed on that org
-        sentry_app = SentryApp.objects.get(
-            installations__organization_id=organization.id,
-            installations__status=SentryAppInstallationStatus.INSTALLED,
-            installations__date_deleted=None,
-            id=sentry_app_id,
-        )
-    except SentryApp.DoesNotExist:
+    # query for the sentry app but make sure it's installed on that org
+    installations = app_service.get_installed_for_organization(organization_id=organization.id)
```

Comment:
Couldn't this run into pagination issues if an organization has many sentry apps installed?

Reply:
Definitely. Is that a real problem? I do think we need to solve pagination generally, we just haven't actually implemented it yet

Reply:
To be fair -- would it be simple enough to have the additional `sentry_app_id` optional filter here?  That prevents the entire class of potential issues.